src/framework/circuit.hpp

```diff
@@ -265,6 +265,13 @@ void Circuit::set_params(bool truncation) {
   size_t last_initialize_pos = 0;
   bool ops_to_remove = false;
 
+  for (size_t i = 0; i < size; ++ i) {
```

Comment:
This adds an unnecessary full iteration over each circuit which is not good for performance. In principle jump operations shouldn't effect truncation, but they may need special handling for how to remap the new locations.